modules/websocket/websocket-destination.c

```diff
@@ -0,0 +1,223 @@
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "websocket-destination.h"
+#include "client.h"
+#include <stdio.h>
+
+
+gboolean
+websocket_dd_init(LogPipe *destination)
+{
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  GlobalConfig *cfg = log_pipe_get_config(destination);
+
+  if (!log_dest_driver_init_method(destination))
+    return FALSE;
+
+  log_template_options_init(&self->template_options, cfg);
+
+  if (self->enable_ssl)
+    self->client_use_ssl_flag = self->allow_self_signed ? 2 : 1;
+  else
+    self->client_use_ssl_flag = 0;
+  return log_threaded_dest_driver_start(destination);
+}
+
+LogDriver *websocket_dd_new(GlobalConfig *cfg);
+
+void websocket_dd_set_port(LogDriver *destination, gint port) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  self->port = port;
+}
+
+void
+websocket_dd_set_address(LogDriver *destination, gchar *address) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  g_free(self->address);
+  self->address = g_strdup(address);
+}
+
+void
+websocket_dd_set_protocol(LogDriver *destination, gchar *protocol) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  g_free(self->protocol);
+  self->protocol = g_strdup(protocol);
+}
+
+void
+websocket_dd_set_path(LogDriver *destination, gchar *path) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  g_free(self->path);
+  self->path = g_strdup(path);
+}
+
+void websocket_dd_set_enable_ssl(LogDriver *destination, int flag) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  self->enable_ssl = flag;
+}
+
+void websocket_dd_set_cert(LogDriver *destination, gchar *path) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  g_free(self->cert);
+  self->cert = path == NULL ? NULL : g_strdup(path);
+}
+
+void websocket_dd_set_key(LogDriver *destination, gchar *path) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  g_free(self->key);
+  self->key = path == NULL ? NULL : g_strdup(path);
+}
+
+void websocket_dd_set_cacert(LogDriver *destination, gchar *path) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  g_free(self->cacert);
+  self->cacert = path == NULL ? NULL : g_strdup(path);
+}
+
+void websocket_dd_set_allow_self_signed(LogDriver *destination, int flag) {
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  self->allow_self_signed = flag;
+}
+
+void
+websocket_dd_set_template(LogDriver *destination, gchar *template)
+{
+  WebsocketDestDriver *self = (WebsocketDestDriver *)destination;
+  GlobalConfig* cfg = log_pipe_get_config(&destination->super);
+  self->template = log_template_new(cfg, NULL);
+  log_template_compile(self->template, template, NULL);
+}
```

Comment:
One of the leak can be found here.
When you say "_new" you don't bother the previous content of "self->template" it leads to leak.